pkg/catalogv2/oci/client.go

```diff
@@ -0,0 +1,316 @@
+	}
+
+	// Create a temp file to store the helm chart tar
+	tempFile, err := os.CreateTemp("", "helm-")
+	if err != nil {
+		return "", false, fmt.Errorf("unable to create temp file for storing the oci artifact")
+	}
+
+	// Check if the oci artifact is of type helm config ?
+	if manifestJSON.ArtifactType == helmregistry.ConfigMediaType || manifestJSON.Config.MediaType == helmregistry.ConfigMediaType {
+		// find the layer of helm chart type and fetch it
+		for _, layer := range manifestJSON.Layers {
+			if layer.MediaType == helmregistry.ChartLayerMediaType {
+				chartTar, err := content.FetchAll(ctx, memoryStore, layer)
+				if err != nil {
+					return "", false, fmt.Errorf("unable to fetch chart blob of %s: %w", ociURL, err)
+				}
+
+				err = os.WriteFile(tempFile.Name(), chartTar, 0o600)
+				if err != nil {
+					return "", false, fmt.Errorf("unable to write chart %s into file %s: %w", ociURL, tempFile.Name(), err)
+				}
+
+				return tempFile.Name(), true, nil
+			}
+		}
+	}
+
+	return tempFile.Name(), false, nil
+}
+
+// getAuthClient creates an oras auth client that can be used
+// in creating an oras registry client or oras repository client.
+func (o *Client) getAuthClient() (http.Client, error) {
+	config := &tls.Config{
+		InsecureSkipVerify: o.insecure,
+	}
+	if len(o.caBundle) > 0 {
+		cert, err := x509.ParseCertificate(o.caBundle)
+		if err != nil {
+			return http.Client{}, err
+		}
+		pool, err := x509.SystemCertPool()
+		if err != nil {
+			logrus.Debugf("getting system cert pool failed: %v", err)
+			pool = x509.NewCertPool()
+		}
+		pool.AddCert(cert)
+
+		config.RootCAs = pool
+	}
+	baseTransport := http.DefaultTransport.(*http.Transport).Clone()
+	baseTransport.TLSClientConfig = config
+
+	return http.Client{
+		Transport: retry.NewTransport(baseTransport),
+	}, nil
+}
+
+// GetOrasRegistry returns the oras registry client along with
+// setting credentials to authenticate with the registry.
+func (o *Client) GetOrasRegistry() (*remote.Registry, error) {
+	orasRegistry, err := remote.NewRegistry(o.registry)
+	if err != nil {
+		return nil, err
+	}
+	orasRegistry.PlainHTTP = o.insecurePlainHTTP
+
+	client, err := o.getAuthClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create an oras auth client: %w", err)
+	}
+
+	orasRegistry.Client = &auth.Client{
+		Cache: auth.DefaultCache,
+		Credential: func(ctx context.Context, reg string) (auth.Credential, error) {
+			return auth.Credential{
+				Username: o.username,
+				Password: o.password,
+			}, nil
+		},
+		Client: &client,
+	}
+
+	return orasRegistry, nil
+}
+
+// GetOrasRepository returns the oras repository client along with
+// setting credentials to authenticate with the registry/repository.
+func (o *Client) GetOrasRepository() (*remote.Repository, error) {
+	orasRepository, err := remote.NewRepository(fmt.Sprintf("%s/%s", o.registry, o.repository))
+	if err != nil {
+		return nil, err
```

Comment:
Can we wrap the error here as well?

Reply:
I purposefully didn't wrap the error since the functions calling `GetOrasRepository` have wrapping and if I wrap this error now, the error message will have duplicated text.